portal/account/account.php

```diff
@@ -15,11 +15,13 @@
 require_once(dirname(__FILE__) . "/../../src/Common/Session/SessionUtil.php");
 OpenEMR\Common\Session\SessionUtil::portalSessionStart();
 
-if ($_SESSION['register'] === true && isset($_SESSION['pid'])) {
+if ($_SESSION['register'] === true && isset($_SESSION['pid']) ||
+    ($_SESSION['credentials_update'] === 1 && isset($_SESSION['pid'])) ||
+    ($_SESSION['itsme'] === 1 && isset($_SESSION['password_update']))) {
```

Comment:
for me:
audit these 2 new conditions

Reply:
Csrf maybe solution here..

Reply:
csrf crossed my mind, but still doesn't address primary issue. Here's an example of a prior related issue OpenEMR has:
https://www.open-emr.org/wiki/index.php?title=Codebase_Security&oldid=27987#Do_not_allow_guessing_of_usernames
(I'm still analyzing this issue)